src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs

```diff
@@ -38,10 +39,19 @@ public void OnBeginRequest(HttpContext httpContext, long timestamp)
         {
             if (this.client.IsEnabled())
             {
-                httpContext.Features.Set(new RequestTelemetry());
+                RequestTelemetry requestTelemetry = new RequestTelemetry();
+                this.client.Initialize(requestTelemetry);
+                requestTelemetry.Id = httpContext.TraceIdentifier;
+                httpContext.Features.Set(requestTelemetry);
 
                 this.beginRequestTimestamp.Value = timestamp;
-                this.client.Context.Operation.Id = httpContext.TraceIdentifier;
+                this.client.Context.Operation.Id = requestTelemetry.Id;
```

Comment:
I just want to make you aware of #343 which is waiting to merge and also makes changes in this block of code. If #343 is merged first you'll almost certainly have a conflict. I can help with the final merge.